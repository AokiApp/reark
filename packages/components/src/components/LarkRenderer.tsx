import React, { useContext } from "react";
import { LarkApiContext } from "../contexts/LarkApiContext";
import { BlockStoreProvider } from "../contexts/BlockStoreContext";
import { BlockComponent } from "./BlockComponent";

export const LarkRenderer: React.FC = () => {
  const { blocks } = useContext(LarkApiContext);

  if (!blocks || blocks.length === 0) return null;

  const rootBlock =
    blocks.find((block) => block.block_type === 1 && !block.parent_id) ||
    blocks[0];

  if (!rootBlock) return null;

  return (
    <BlockStoreProvider items={blocks}>
      <BlockComponent blockId={rootBlock.block_id} />
    </BlockStoreProvider>
  );
};
