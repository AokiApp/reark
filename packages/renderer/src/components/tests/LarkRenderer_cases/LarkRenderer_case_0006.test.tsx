import { render } from "@testing-library/react";
import { LarkRenderer } from "../../LarkRenderer";

describe("LarkRenderer 異常系: textプロパティ欠損", () => {
  it("textプロパティが存在しないブロックもレンダリングできる", () => {
    const block = {
      block_id: "no-text",
      block_type: 2,
      parent_id: "",
      children: [],
      // textプロパティ無し
    };
    const blocks = [block];
    const { container } = render(<LarkRenderer initialData={{ blocks }} />);
    expect(container).toMatchSnapshot();
  });
});
