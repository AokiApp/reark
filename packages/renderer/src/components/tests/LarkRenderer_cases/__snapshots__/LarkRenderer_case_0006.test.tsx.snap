// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LarkRenderer 異常系: textプロパティ欠損 textプロパティが存在しないブロックもレンダリングできる 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div>
      <h1>
        Something went wrong.
      </h1>
      <p>
        We are working to fix this issue. Please try again later.
      </p>
    </div>
  </div>
</div>
`;
